### ISSUE TITLE
autogen external link chause a wrong linting

### ISSUE THREAD

**Issue by rebortg** [vale-cli/vale#163]
Hi,

i found a strange issue while linting RST files.
https://raw.githubusercontent.com/vyos/vyos-documentation/master/docs/clustering.rst

I want to use a  substitution
```
swap:
  vyos: VyOS
```

The code block should actually be ignored. But with the automatically generated link (on line 11) the first match in the code-block will be found and marked as a warning. The rest will be ignored. When i change the the URL in a RST format like ``` `T985 <https://phabricator.vyos.net/T985/>`_ ``` all went fine and the linter worked as it should.

i `diff` in the generated html code, there is no diff
autogen:
```
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Please follow the process of the cluster function here. <a class="reference external" href="https://phabricator.vyos.net/T985">https://phabricator.vyos.net/T985</a></p>
</div>
````

with 'correct' syntax:
```
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Please follow the process of the cluster function here. <a class="reference external" href="https://phabricator.vyos.net/T985">https://phabricator.vyos.net/T985</a></p>
</div>
```

OS (e.g., macOS, Windows, Linux, etc.)
- MacOS
- Docker image (ubuntu)

Install method
 - direct download

Vale version
- 1.7.1
- 2.0.0 beta2

**Comment by jdkato** [comment]
Thanks for the report -- I'm going to be doing a fairly significant refactor to how Sphinx files are handled (see #147) for the next release and I expect these edge cases to be fixed.

**Comment by jdkato** [comment]
This should be fixed in [`v2.0.0-beta.3`](https://github.com/errata-ai/vale/releases/tag/v2.0.0-beta.3).
